### PR TITLE
Issue 9275 - GeoJsonDataSource `process` method

### PR DESCRIPTION
**Pull request by thw0rted**
https://github.com/CesiumGS/cesium/issues/9275

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @thw0rted!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/CesiumGS/cesium/blob/issue-9275/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

**Comment by thw0rted** [comment]
About half of this PR is fixes for unspecified or inconsistent JSDoc -- if I need to break that out separately I can, but since I was already touching the code in question I figured it wouldn't hurt.

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @thw0rted! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by hpinkos** [comment]
Thanks for the PR @thw0rted.  Sorry we haven't had a chance to review this yet.  I skimmed through the changes and they look fine.  I'll try to make time to give this a thorough review soon.

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @thw0rted! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @thw0rted! 

No one has commented on this pull request in 90 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 90 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @thw0rted! 

No one has commented on this pull request in 90 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 90 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @thw0rted! 

No one has commented on this pull request in 90 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 90 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

File: .vscode/tasks.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (sanjeetsuhag) [inline .vscode/tasks.json:1]: We should take a holistic look at all our other tasks and see how they can be better integrated with VSCode in a separate PR. For now, this file should be removed.

File: CHANGES.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 ### 1.77 - 2020-01-04
 
+##### Additions :tada:
+
+- Added `GeoJsonSource.process` to support adding features without removing existing entities, similar to `CzmlDataSource.process`. [#9275](https://github.com/CesiumGS/cesium/issues/9275)
```
Comment (sanjeetsuhag) [inline CHANGES.md:7]: This needs to be updated.

**Review by sanjeetsuhag — COMMENTED** [review]
Thanks for the contribution @thw0rted. This PR looks mostly in good shape. A few minor changes will get it over the finish line.

**Comment by thw0rted** [comment]
Oh wow, I totally forgot about this -- I'll have to dig through my notes and try to remember why I needed  this 😁 

I hope it's OK that I rebased with force-push rather than a merge commit, that's how I usually work locally.

Re: vscode files, I noticed that gitignore already has entries for WebStorm, would it make sense to add one for `.vscode/` before closing this out?

Also also: I have no idea how out of date #8903 is but while I have your attention, maybe somebody could take a look?

**Comment by ggetz** [comment]
> I have no idea how out of date https://github.com/CesiumGS/cesium/pull/8903 is but while I have your attention, maybe somebody could take a look?

@thw0rted We're a bit bottled-necked in terms of devs with enough typescript knowledge, but we'll try to get to it soon. Thanks for your patience!

File: Source/DataSources/CzmlDataSource.js [hunk 3]
```diff
@@ -4985,10 +4985,23 @@ Object.defineProperties(CzmlDataSource.prototype, {
   },
 });
 
+// TODO: typedef for Packet (and about a hundred other things)
```
Comment (sanjeetsuhag) [inline Source/DataSources/CzmlDataSource.js:4988]: Is there still some TODO items remaining on this?

File: .vscode/tasks.json [hunk 4]
```diff
@@ -1,27 +0,0 @@
-{
```
Comment (sanjeetsuhag) [inline .vscode/tasks.json:1]: An update from `main` should fix this. We have `tasks.json` by default now.

**Review by sanjeetsuhag — COMMENTED** [review]
Just revisiting this PR, seems like it's close to a merge, but needs an update pass. Also, there are some unrelated changes here in the form of TypeScript changes but it would be preferred if they can be put in a dedicated PR like #8903. The `gulpfile.cjs` change is probably ok.

**Comment by thw0rted** [comment]
I've merged the latest from `main` and I think addressed all the comments above.  Please let me know if anything else needs to be changed, or feel free to make the fix yourself, whatever is easiest.

**Comment by thw0rted** [comment]
FYI I went ahead and updated #8903 as well, since you're looking at this.  There are a bunch of what turned out to be trivial fixes due to the `when.js` deprecation. I fixed `@returns {Promise<void>}` to `@returns {when.Promise.<void>}`, since the dotted-generic notation is technically correct in JSDoc, then when `when` was removed I replaced `when.Promise` with `Promise`, so you're left with a bunch of file changes that add a single `.`.  If there's an easy way to remove all those changes from the PR, let me know, I'm still looking for it.

**Comment by thw0rted** [comment]
I'm confused, it looks like `main` has a `.vscode/tasks.json` and your commit just deleted that file.  Was this intentional?

**Comment by sanjeetsuhag** [comment]
@thw0rted Good catch. I deleted it because in `main`, the file is in [`.vscode/tasks.json`](https://github.com/CesiumGS/cesium/blob/main/.vscode/tasks.json). The file in this PR was in the root directory. 

image

But now, looking at the PR diff, it's deleting it inside `.vscode/tasks.json`...

**Review by sanjeetsuhag — APPROVED** [review]
Using this [Sandcastle](http://localhost:8080/Apps/Sandcastle/index.html#c=dVFdawIxEPwrS57uoOSqr55SUCiUQh+kfQpIzK2aNiaSD8WW/vdu7qCn9Qphk83sDJmJcjZEOGo8oYcpWDzBHINOe/7W3hWCqbafOxultugFKyfCCqtaZiOjXLrkFb56MyIFwTivaC3l/mBwQXAVdD6GeKbCt+jeg7OCTQYlxsMSFlej+/0qhVWIMpJMdIde51bp2sojuieaXfzCRbbQmeY9KXDZNEXfdz77nhsnL/HsuORxh7bYJKuidhaKEr6EhYun8IN3pB2uieP/iAC5fpcT2qmyO1a3yc06EOCB0nQ+QvKmoJAiUkg5kmqd1AdGrkLID8+jdXVJrRt9BN1MB34UlJEhELJJxiz1Jwo2qyuav6HmCLTdvhzRG3nOY7vR7Lm75JzXFbXDzOicWUv/R/kH), I confirmed that this was working as expected with two GeoJSONs loading, first with `load`, then with `process`.

image

Thanks again @thw0rted!
